src/haar.jl

```diff
@@ -0,0 +1,159 @@
+"""
+```
+features = haar_features(img, top_left, bottom_right, feat)
+features = haar_features(img, top_left, bottom_right, feat, coordinates)
+```
+
+Returns an array containing the Haar-like features for the given Integral Image in the region specified by the points top_left and bottom_right.
+
+Parameters:
+
+ -  img               = The Integral Image for which the Haar-like features are to be found
+ -  top_left          = The top and left most point of the region where the features are to be found
+ -  bottom_right      = The bottom and right most point of the region where the features are to be found
+ -  feat              = A symbol specifying the type of the Haar-like feature to be found
+
+```
+    Currently, feat can take 5 values:
```

Comment:
How about, instead of using symbols to enumerate different combinations of horizontal or vertical rectangles, you just take a parameter for the number of rectangles a user wishes to have along the horizontal axis, and a second parameter for the number of rectangles the user wishes to have along the vertical axis?

Reply:
Since there are only 5 combinations to used in this case, taking 2 separate parameters would be unnecessary as the user will have only these 5 combinations to choose from:

`x = 2 y = 1`
`x = 1 y = 2`
`x = 3 y = 1`
`x = 1 y = 3`
`x = 2 y = 2`

So only single parameter for selecting these should suffice i guess.

Reply:
Now that I understand better how you implemented it, i.e. that you actually extract features for all possible sizes for a particular rectangle configuration, I think it makes more sense  to discard my suggestion for parameters specifying the number of rectangles and leave things as they are. However, perhaps we can  give more descriptive names to the symbols.

Also, it may be useful to have an ascii drawing similar to what I posted in a comment below to explain what is meant by two rectangles along horizontal or vertical axis. The xy4 one in particular could be misunderstood. I understand it to mean that the rectangles are laid out in a 2x2 grid. 

It is also worth explaining which rectangles  are subtracted and which are added to evaluate the final haar feature.